Tests/Clappr_Tests/PlayButtonTests.swift

```diff
@@ -0,0 +1,244 @@
+                    }
+
+                    it("calls the playback play") {
+                        playButton.button.sendActions(for: .touchUpInside)
+
+                        expect(coreStub.playbackMock?.didCallPlay).to(beTrue())
+                    }
+
+                    it("changes the image to a pause icon") {
+                        let pauseIcon = UIImage.from(name: "pause")!
+
+                        playButton.button.sendActions(for: .touchUpInside)
+
+                        let currentButtonIcon = (playButton.button.imageView?.image)!
+                        expect(currentButtonIcon.isEqualTo(image: pauseIcon)).toEventually(beTrue())
+                    }
+
+                    context("and is vod") {
+                        it("shows button") {
+                            let coreStub = CoreStub()
+                            let playButton = PlayButton(context: coreStub)
+                            playButton.render()
+                            playButton.view.isHidden = true
+
+                            coreStub.activePlayback?.trigger(Event.didPause.rawValue)
+
+                            expect(playButton.view.isHidden).to(beFalse())
+                        }
+                    }
+                }
+            }
+
+            describe("render") {
+
+                it("set's acessibilityIdentifier to button") {
+                    let playButton = PlayButton()
+
+                    playButton.render()
+
+                    expect(playButton.button.accessibilityIdentifier).to(equal("PlayPauseButton"))
+                }
+
+                describe("button") {
+                    it("adds it in the view") {
+                        let playButton = PlayButton()
+
+                        playButton.render()
+
+                        expect(playButton.view.subviews).to(contain(playButton.button))
+                    }
+
+                    it("has scaleAspectFit content mode") {
+                        let playButton = PlayButton()
+
+                        playButton.render()
+
+                        expect(playButton.button.imageView?.contentMode).to(equal(UIViewContentMode.scaleAspectFit))
+                    }
+                }
+
+            }
+
+            context("when stalled") {
+                it("hides the plugin") {
+                    let coreStub = CoreStub()
+                    let playButton = PlayButton(context: coreStub)
+
+                    coreStub.activePlayback?.trigger(Event.stalled.rawValue)
+
+                    expect(playButton.view.isHidden).to(beTrue())
+                }
+
+                it("hides the plugin") {
+                    let coreStub = CoreStub()
+                    let playButton = PlayButton(context: coreStub)
+
+                    coreStub.activePlayback?.trigger(Event.playing.rawValue)
+
+                    expect(playButton.view.isHidden).to(beFalse())
+                }
+            }
+        }
+    }
+}
+
+extension UIImage {
```

Comment:
Don't u think it is better to extract this extension to another file?

Reply:
Done.